### PR TITLE
Fix #3144: Drop disappear on blur in MaskedInput

### LINKED ISSUE #3144: MaskedInput - cannot exit dropdown without selecting value
Once the MaskedInput is selected, the dropdown does not disappear until a value is selected. If user clicks somewhere else on the page, the dropdown remains active which is not expected behavior. 

### Expected Behavior

When user moves away from the MaskedInput field, the dropdown should disappear.

### Actual Behavior

Dropdown stays visible and does not go away until user enters a valid value.

### URL, screen shot, or Codepen exhibiting the issue

This behavior can be seen on the grommet docs code sample https://v2.grommet.io/maskedinput

### PR DESCRIPTION
**Pull request by shivamsingha**
#### What does this PR do?
Makes drop disappear on blur in MaskedInput
#### Where should the reviewer start?
src/js/components/MaskedInput/MaskedInput.js
#### What testing has been done on this PR?
Storybook
#### How should this be manually tested?
Storybook

#### What are the relevant issues?
#3144 
#### Do the grommet docs need to be updated?
No
#### Should this PR be mentioned in the release notes?
Yes
#### Is this change backwards compatible or is it a breaking change?
Backwards compatible

### CODE REVIEW

File: src/js/components/MaskedInput/MaskedInput.js [hunk 1]
```diff
@@ -122,7 +122,7 @@ class MaskedInput extends Component {
     return null;
   }
 
-  state = {};
+  state = { focused: false };
```
Comment (crashuniverse) [inline src/js/components/MaskedInput/MaskedInput.js]: Maybe `focussed`?

**Comment by ericsoderberghp** [comment]
Agreed that the issue needs fixing. Thanks for working on it. Unfortunately, while the proposed changes seem to be working well on Chrome, they seem to be a step backwards on Safari and Firefox. For both of those browsers, clicking something in the drop down doesn't result it in being added to the input. I think there is another pre-existing problem in Firefox where the item in the drop must be clicked twice before it is selected. But, with these changes, even that doesn't work.

**Comment by shivamsingha** [comment]
@ericsoderberghp  Sorry but unfortunately I have only Linux so can't test on Safari.  

In case of Firefox, it's working fine for me in v67. Which version are you having problems in?

gif

**Comment by ericsoderberghp** [comment]
Here's what I see on Safari on OSX, same with Firefox as well:
ezgif com-optimize

**Comment by oorestisime** [comment]
Hm weird. i dont have this on FF. which FF version?

**Comment by ericsoderberghp** [comment]
Hmm... It appears to be sporadic for me. Sometimes, clicking on the drop down options don't work, no mater how many times I try. Sometimes, the first one works but then the second drop down doesn't show unless I click away from the component and then focus again.

I'm on the latest Firefox.

**Comment by shivamsingha** [comment]
@ericsoderberghp Did you check the React devtools if the states are changing? I didn't change anything drastic.   
   
I'm also on the latest Firefox, the stable one. It's works on the Nightly channel, version 69.0a1.
Are you on mac?

**Comment by elieteyssedou** [comment]
@ericsoderberghp Any news on this ? Would be great to fix this 🙏

**Comment by ericsoderberghp** [comment]
Here's a version of MaskedInput.js that is a bit simpler, removing `focused` from the state entirely, allows `Esc` to close the drop while retaining focus, and seems to behave better overall.

```

**Comment by shivamsingha** [comment]
@ericsoderberghp Thanks. It's working great.

**Comment by ericsoderberghp** [comment]
Thanks for your contribution!

**Pull request merged**